tiiuae/ghaf
# labwc: Replace waybar with ironbar

Replaces waybar with ironbar.

[Ironbar] is a replacement for waybar that is customisable, modular, and supports showing running applications on the taskbar.

Depends on: #582

[Ironbar]: https://github.com/JakeStanger/ironbar 

## Description of changes

- Replaced waybar with ironbar, preserving same style.
- Running applications now show on taskbar, can switch tasks.
- Added volume widget.
  - Ability to change master volume, and individual VMs volume.
- Align more with Figma designs.

Also other changes:

- Added window shadows
- Fixes wifi icon (used to show code when percentage <30)
- Launcher now opens instantly
- Volume media keyboard hotkeys now change volume instantly.

## Known issues

- ~If too many windows are open, or with long window names, taskbar may go off screen (to the right).~
- ~Sometimes when closing the launcher, the compositor won't update until you hover over it. Seems to only happen `pixman` (software-based) renderer.~
- ~Moving audio slider doesn't change volume, but it does correctly update when using the volume media keys.~
- ~Two taskbars are shown when an external display is connected.~
- Battery icon in taskbar uses wrong theme (light theme, dark icon over dark panel).
- Battery icon shows on Orin (0% with question mark battery icon).

## Checklist for things done






## Testing

- Open applications, they should show up on the taskbar.
- Window should now have a border shadow.
- You can switch by clicking on the launcher on the taskbar.
- App drawer works.
  - Should now launch almost instantly, delay should be gone.
  - Quickly opening and closing shouldn't cause the launcher to crash irrecoverably.
- Clicking on WiFi icon shows SSID and IP.
  - Test with different wifi strengths.
  - When signal is less than 30%, it should show correct icon (bug fixed).
- Clicking on volume shows the volume slider, works (test with video playing).
  - You can individually turn down volume of a single VM.
  - Clicking on volume icon inside of the pop-up would toggle mute.
  - Volume up, down, and mute keys reflect instantly on slider and changes volume immediately.
- Clicking on battery while discharging/charging will show estimate to empty/full.
- Clicking on settings cog shows network settings application.
- Clicking on time/date shows mini calendar.
- Connecting with external display works as expected without issues, disconnecting also shouldn't cause any issues.
- Open browser tab with long name, the taskbar item name should be truncated (with ellipsis).
- Open many applications, fill taskbar. Taskbar should not go off-screen.
